still getting issues on cargo run:

```
➜  jade-signer-rpc git:(issue/hd_wallet) cargo run
   Compiling jade_signer_rs v0.1.0 (/Users/shanejonas/scratch/jade-signer-rpc/jade-signer-rs)
error[E0432]: unresolved import `hdwallet`
 --> jade-signer-rs/src/rpc/error.rs:6:5
  |
6 | use hdwallet;
  |     ^^^^^^^^ no `hdwallet` in the root

error[E0432]: unresolved import `mnemonic::HDPath`
  --> jade-signer-rs/src/rpc/serves.rs:19:35
   |
19 | use mnemonic::{self, gen_entropy, HDPath, Language, Mnemonic, ENTROPY_BYTE_LENGTH};
   |                                   ^^^^^^ no `HDPath` in `mnemonic`

error[E0433]: failed to resolve: use of undeclared type or module `SerializableKeyFileHD`
   --> jade-signer-rs/src/keystore/serialize/mod.rs:172:29
    |
172 |             Err(_) => match SerializableKeyFileHD::try_from(self) {
    |                             ^^^^^^^^^^^^^^^^^^^^^ use of undeclared type or module `SerializableKeyFileHD`

error[E0425]: cannot find function `generate_key` in module `mnemonic`
   --> jade-signer-rs/src/rpc/serves.rs:610:24
    |
610 |     let pk = mnemonic::generate_key(&hd_path, &mnemonic.seed(""))?;
    |                        ^^^^^^^^^^^^ not found in `mnemonic`

warning: unused import: `std::cell::RefCell`
  --> jade-signer-rs/src/rpc/serves.rs:21:5
   |
21 | use std::cell::RefCell;
   |     ^^^^^^^^^^^^^^^^^^
   |
   = note: #[warn(unused_imports)] on by default

warning: unused import: `std::cell::RefCell`
  --> jade-signer-rs/src/rpc/mod.rs:21:5
   |
21 | use std::cell::RefCell;
   |     ^^^^^^^^^^^^^^^^^^

warning: use of deprecated item 'std::env::home_dir': This function's behavior is unexpected and probably not what you want. Consider using the home_dir function from https://crates.io/crates/dirs instead.
  --> jade-signer-rs/src/storage/mod.rs:42:26
   |
42 |     let mut config_dir = env::home_dir().expect("Expect path to home dir");
   |                          ^^^^^^^^^^^^^
   |
   = note: #[warn(deprecated)] on by default

error[E0599]: no variant named `HdWallet` found for type `keystore::CryptoType` in the current scope
  --> jade-signer-rs/src/storage/keyfile/mod.rs:62:25
   |
62 |             CryptoType::HdWallet(_) => true,
   |             ------------^^^^^^^^--- variant not found in `keystore::CryptoType`
   |
  ::: jade-signer-rs/src/keystore/mod.rs:66:1
   |
66 | pub enum CryptoType {
   | ------------------- variant `HdWallet` not found here

error: aborting due to 5 previous errors

Some errors occurred: E0425, E0432, E0433, E0599.
For more information about an error, try `rustc --explain E0425`.
error: Could not compile `jade_signer_rs`.

To learn more, run the command again with --verbose.
```

--- a/ci/install_rust.sh
+++ b/ci/install_rust.sh
@@ -6,8 +6,6 @@ set -e
 # Print all executed commands
 set -x
 
-apt-get install -y libusb-1.0-0-dev
-
 # Download rustup script and execute it
 curl https://sh.rustup.rs -sSf > ./rustup.sh
 chmod +x ./rustup.sh

--- a/jade-signer-rs/src/keystore/mod.rs
+++ b/jade-signer-rs/src/keystore/mod.rs
@@ -67,8 +67,8 @@ pub enum CryptoType {
     /// normal Web3 Secret Storage
     Core(CoreCrypto),
 
-    #[cfg(feature = "hardware-wallet")]
     /// backed with HD Wallet
+    #[cfg(feature = "hardware-wallet")]
     HdWallet(HdwalletCrypto),
 }
 

--- a/jade-signer-rs/src/keystore/serialize/mod.rs
+++ b/jade-signer-rs/src/keystore/serialize/mod.rs
@@ -66,9 +66,9 @@ impl Into<KeyFile> for SerializableKeyFileCore {
     }
 }
 
-#[cfg(feature = "hardware-wallet")]
 /// A serializable keystore file (UTC / JSON format)
 #[derive(Serialize, Deserialize, Clone, Debug)]
+#[cfg(feature = "hardware-wallet")]
 pub struct SerializableKeyFileHD {
     version: u8,
     id: Uuid,
@@ -163,6 +163,18 @@ impl KeyFile {
 }
 
 impl Serialize for KeyFile {
+    #[cfg(feature = "default")]
+    fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
+    where
+        S: Serializer,
+    {
+        match SerializableKeyFileCore::try_from(self.clone()) {
+            Ok(sf) => sf.serialize(serializer),
+            Err(e) => Err(ser::Error::custom(e)),
+        }
+    }
+
+    #[cfg(feature = "hardware-wallet")]
     fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
     where
         S: Serializer,

--- a/jade-signer-rs/src/rpc/error.rs
+++ b/jade-signer-rs/src/rpc/error.rs
@@ -3,6 +3,7 @@
 use super::core;
 use super::storage;
 use contract;
+#[cfg(feature = "hardware-wallet")]
 use hdwallet;
 use hex;
 use jsonrpc_core;
@@ -97,6 +98,7 @@ impl From<mnemonic::Error> for Error {
     }
 }
 
+#[cfg(feature = "hardware-wallet")]
 impl From<hdwallet::Error> for Error {
     fn from(err: hdwallet::Error) -> Self {
         Error::MnemonicError(err.to_string())

--- a/jade-signer-rs/src/rpc/mod.rs
+++ b/jade-signer-rs/src/rpc/mod.rs
@@ -18,7 +18,6 @@ use log::Level;
 use serde::de::DeserializeOwned;
 use serde::Serialize;
 use serde_json::{self, Value};
-use std::cell::RefCell;
 use std::net::SocketAddr;
 use std::sync::{Arc, Mutex};
 
@@ -216,6 +215,7 @@ pub fn start(addr: &SocketAddr, storage_ctrl: StorageController, sec_level: Opti
         });
     }
 
+    #[cfg(feature = "hardware-wallet")]
     {
         let storage_ctrl = Arc::clone(&storage_ctrl);
         io.add_method("signer_importMnemonic", move |p: Params| {

--- a/jade-signer-rs/src/rpc/serves.rs
+++ b/jade-signer-rs/src/rpc/serves.rs
@@ -15,10 +15,11 @@ use hdwallet::bip32::to_prefixed_path;
 #[cfg(feature = "hardware-wallet")]
 use hdwallet::WManager;
 use jsonrpc_core::{Params, Value};
-use keystore::{os_random, CryptoType, Kdf, KdfDepthLevel, KeyFile, PBKDF2_KDF_NAME};
-use mnemonic::{self, gen_entropy, HDPath, Language, Mnemonic, ENTROPY_BYTE_LENGTH};
+use keystore::{CryptoType, KdfDepthLevel, KeyFile};
+#[cfg(feature = "hardware-wallet")]
+use mnemonic::HDPath;
+use mnemonic::{gen_entropy, Language, Mnemonic, ENTROPY_BYTE_LENGTH};
 use serde_json;
-use std::cell::RefCell;
 use std::ops::Deref;
 use std::str::FromStr;
 use std::sync::{Arc, Mutex};
@@ -592,6 +593,7 @@ pub fn generate_mnemonic() -> Result<String, Error> {
     Ok(mnemonic.sentence())
 }
 
+#[cfg(feature = "hardware-wallet")]
 pub fn import_mnemonic(
     params: Either<(NewMnemonicAccount,), (NewMnemonicAccount, CommonAdditional)>,
     sec: &KdfDepthLevel,

--- a/jade-signer-rs/src/storage/keyfile/mod.rs
+++ b/jade-signer-rs/src/storage/keyfile/mod.rs
@@ -40,6 +40,31 @@ pub struct AccountInfo {
     pub is_hidden: bool,
 }
 
+#[cfg(feature = "default")]
+impl From<KeyFile> for AccountInfo {
+    fn from(kf: KeyFile) -> Self {
+        let mut info = Self::default();
+        info.address = kf.address.to_string();
+
+        if let Some(name) = kf.name {
+            info.name = name;
+        };
+
+        if let Some(desc) = kf.description {
+            info.description = desc;
+        };
+
+        if let Some(visible) = kf.visible {
+            info.is_hidden = !visible;
+        };
+
+        info.is_hardware = false;
+
+        info
+    }
+}
+
+#[cfg(feature = "hardware-wallet")]
 impl From<KeyFile> for AccountInfo {
     fn from(kf: KeyFile) -> Self {
         let mut info = Self::default();

